### PR TITLE
Pass options to error integration

### LINKED ISSUE #746: [2.0] error_types option doesn't work
When I set option error_types to (E_ALL & ~E_WARNING & ~E_NOTICE) I expect that types E_WARNING and E_NOTICE won't send information to server.

Code example:
```
<?php
require(__DIR__ . '/vendor/autoload.php');
use function Sentry\init;
init(['dsn' => __DSN___, 'error_types' => E_ALL & ~E_WARNING & ~E_NOTICE]));
$my_array = array();
$info_error= $my_array[2];
```

But when I execute it I receive:
`Notice: Undefined offset: 2`

It happened because of `capturedErrors` private property in ErrorHandler class.
https://github.com/getsentry/sentry-php/blob/master/src/ErrorHandler.php#L46

Because of that `shouldCaptureError` is always true.
https://github.com/getsentry/sentry-php/blob/master/src/ErrorHandler.php#L181

---
php version 7.1.24
sentry package 2.0.0-beta1

### PR DESCRIPTION
**Pull request by HazAT**
Fixes #746

I hope I got the comparison right 🙈 
https://github.com/getsentry/sentry-php/compare/fix/error_types-option?expand=1#diff-9a950f4db299479d9933d02a78100059R40

### CODE REVIEW

**Comment by ste93cry** [comment]
I totally disagree with the solution implemented in this PR, we are leaking part of the purpose of the error handler into an integration just because the current architecture has some limitations in this specific case. I would like to think a bit more about it since I'm working on a different solution in my free time

**Comment by HazAT** [comment]
@ste93cry I do not fully understand your concerns.
We could probably remove the part from the `ErrorHandler` that lets you set which error it should handle.
I would make it that the `ErrorHandler` always receives all errors. 

Yes, the `ErrorHandler` should be a singleton, since dealing with multiple `set_error_handler` etc. is very complex so having one class responsible for it is enough.

Additionally what we could do is, instead of having `ErrorHandler::register` we could have something like `ErrorHandler::addCallback` so basically an array of "listeners". In case of an error the `ErrorHandler` calls all listeners with the error and level.

It's up to the listener (in our case the `ErrorHandlerIntegration`) to decide what to do with this error.

I think this approach is very straight forward and easy, we do similar stuff in other SDKs already.

File: tests/phpt/send_only_warning.phpt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+]);
+
+error_reporting(E_ALL);
+
+$client = Hub::getCurrent()->getClient();
+
+/** @var \Sentry\Transport\HttpTransport $transport */
+$transport = Assert::getObjectAttribute($client, 'transport');
+
+trigger_error("Cannot divide by zero", E_USER_ERROR);
```
Comment (Jean85) [inline tests/phpt/send_only_warning.phpt]: This test is wrong: no code below this point is ever executed, this error halts execution.

File: src/ErrorHandler.php [hunk 2]
```diff
@@ -41,10 +41,15 @@ final class ErrorHandler
...
     /**
-     * @var int The errors that will be catched by the error handler
+     * @var int The errors that will be caught by the error handler
      */
     private $capturedErrors = E_ALL;
 
+    /**
+     * @var int The last caught error type
+     */
+    private $lastErrorType = 0;
```
Comment (Jean85) [inline src/ErrorHandler.php]: Why using a property and not passing it as every invocation of `handleException`?

**Pull request closed without merging**